File: api/src/main/java/com/prep/api/model/imagem/Imagem.java
package com.prep.api.model.imagem;

import com.prep.api.model.ModelAbstract;

public class Imagem extends ModelAbstract {

    protected Long id, tamanho;
    protected String nome, tipo;
    protected byte[] dados;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getTamanho() {
        return tamanho;
    }

    public void setTamanho(Long tamanho) {
        this.tamanho = tamanho;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public byte[] getDados() {
        return dados;
    }

    public void setDados(byte[] dados) {
        this.dados = dados;
    }
}

File: ui/src/app/modules/pessoa/models/imagem.ts
export class Imagem {

  public id: number;
  public tamanho: number;
  public nome: string;
  public tipo: string;
  public dados;

  public static getImagemBase64(imagem: Imagem, alternative: string): string {
    if (imagem && imagem.dados) {
      return `data:${imagem.tipo};base64,${imagem.dados}`;
    } else {
      return alternative;
    }
  }

  public setImagemApi(imagem) {
    this.id = imagem.id;
    this.dados = imagem.dados;
    this.nome = imagem.nome;
    this.tamanho = imagem.tamanho;
    this.tipo = imagem.tipo;
  }
}
